src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlRegexTranslator.cs

```diff
@@ -0,0 +1,224 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Text.RegularExpressions;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal;
+using ExpressionExtensions = Microsoft.EntityFrameworkCore.Query.ExpressionExtensions;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal;
+
+/// <summary>
+///     Translates Regex method calls into their corresponding PostgreSQL equivalent for database-side processing.
+/// </summary>
+/// <remarks>
+///     http://www.postgresql.org/docs/current/static/functions-matching.html
+/// </remarks>
+public class NpgsqlRegexTranslator : IMethodCallTranslator
+{
+    private static readonly MethodInfo IsMatch =
```

Comment:
FYI we're starting to move away from this way of matching things in method/member translators; this lookup of MethodInfos via reflection causes them to always get preserved when using trimming, regardless of whether the user actually needs them or not (and therefore bloats the final binary). Instead, it's better to check the declaring type, and then simply check the method name as a string.

I'll rewrite to do this.